keps/sig-cloud-provider/2395-removing-in-tree-cloud-providers/README.md

```diff
@@ -169,6 +192,151 @@ import (
 )
 ```
 
+## Production Readiness Review Questionnaire
+
+### Feature Enablement and Rollback
+
+_This section must be completed when targeting alpha to a release._
+
+* **How can this feature be enabled / disabled in a live cluster?**
+  - [X] Feature gate (also fill in values in `kep.yaml`)
+    - Feature gate name: DisableCloudProviders
+    - Components depending on the feature gate: kubelet, kube-apiserver, kube-controller-manager
+  - [X] Feature gate (also fill in values in `kep.yaml`)
+    - Feature gate name: DisableKubeletCloudCredentialProvider
+    - Components depending on the feature gate: kubelet
+
+* **Does enabling the feature change any default behavior?**
+  Yes, enabling this feature will disable all capabilities enabled when --cloud-provider is set.
+  Users need to ensure they have migrated to out-of-tree components prior to enabling this feature gate.
+
+* **Can the feature be disabled once it has been enabled (i.e. can we roll back
+  the enablement)?**
+  Yes, the feature can be disabled once it is enabled. If disabled, users must ensure
+  any out-of-tree counterpart components are no longer running in the cluster.
+
+* **What happens if we reenable the feature if it was previously rolled back?**
+
+  All capabilities from in-tree cloud providers will be re-disabled.
+
+* **Are there any tests for feature enablement/disablement?**
+  Adequate unit tests, component integration test and e2e tests will be added for this feature before
```

Comment:
Fundamentally we are relying on the in-tree cloud providers to run e2e & conformance tests with their platform. For managed solutions I believe that is literally the only way for this to work. For the roll your own on those platforms its fair point but I suspect what we'll find is that our tests will just fail. (Eg. I thing any windows node tests on GCE will just fail due to the aforementioned CIDR issue)

Reply:
ack -- I think transition into beta will also force us to update our e2es to use external components